bgpd/bgp_network.c

```diff
@@ -153,40 +155,48 @@ static int bgp_md5_set_password(struct peer *peer, const char *password)
 	struct listnode *node;
 	int ret = 0;
 	struct bgp_listener *listener;
+	struct bgp *bgp = NULL;
 
-	/*
-	 * Set or unset the password on the listen socket(s). Outbound
+	/* Set or unset the password on the listen socket(s). Outbound
 	 * connections are taken care of in bgp_connect() below.
+	 *
+	 * Walk through all the Listener sockets and find the one
+	 * corresponding to the BGP instance of the given peer
 	 */
-	frr_with_privs(&bgpd_privs) {
-		for (ALL_LIST_ELEMENTS_RO(bm->listen_sockets, node, listener))
-			if (listener->su.sa.sa_family ==
-			    peer->su.sa.sa_family) {
-				uint16_t prefixlen =
-					peer->su.sa.sa_family == AF_INET
-					? IPV4_MAX_PREFIXLEN
-					: IPV6_MAX_PREFIXLEN;
+	for (ALL_LIST_ELEMENTS_RO(bm->listen_sockets, node, listener)) {
+		if (listener->su.sa.sa_family == peer->su.sa.sa_family) {
 
-				/*
-				 * if we have stored a BGP vrf instance in the
-				 * listener it must match the bgp instance in
-				 * the peer otherwise the peer bgp instance
-				 * must be the default vrf or a view instance
-				 */
-				if (!listener->bgp) {
-					if (peer->bgp->vrf_id != VRF_DEFAULT
-					    && peer->bgp->inst_type
-						       != BGP_INSTANCE_TYPE_VIEW)
-						continue;
-				} else if (listener->bgp != peer->bgp)
-					continue;
+			/* Obtain BGP instance corresponding to a specific listener socket
+			 * - if it is a VRF netns sock, then BGP is in listener structure
+			 * - otherwise, the bgp instance need to be demultiplexed
```

Comment:
This is incorrect - the listener->bgp is set for all non default vrfs. Is this an old fix that has been rebased onto master? I don't think it is actually fixing anything since the problem in the description was already addressed.